### ISSUE TITLE
Quick Dial is lagging

### ISSUE THREAD

**Issue by SultanRahi** [MatMoul/quickdial-webext#81]
There's a lagging problem with Quick Dial since few days (new update ?).
First, it just froze everything. So after restarting two or three times Firefox, I reinstalled it.
Now it's working but I have to wait few seconds before it responds.
Same for the options.

It's just too slow and unusable.

What went wrong ?

By the way, another person, who wrote a message on the add-on web page seems to have the same problem.

**Comment by MatMoul** [comment]
Thank you for your return, the method I use seem to be too expensive with too many bookmark.
I'll work on it when I find time.
This issue can be related to #70

**Comment by XSkaarj** [comment]
Same thing here. So slow it's unusable. I do love this add-on, matmoul. Thanks for your work and hope to be using it again soon

**Comment by XSkaarj** [comment]
After a Firefox update this morning it's no longer lagging. Firefox 58.0.2

**Comment by MatMoul** [comment]
@XSkaarj Good to heart that but I need to work on the data...

**Comment by vertigo220** [comment]
I'm using Waterfox 56.1.0 x64 and whenever I open a new tab, I have to wait a few seconds before I can use the browser normally. Typically, when using a blank tab for new tabs, I open one and can immediately start typing and it works as expected, but with QD, if I try typing right away it's very laggy and unresponsive. I only have 9 bookmarks on it, no folders, and this is on W10 x64 with an i7-7820x and 64GB of RAM, so it definitely shouldn't be a hardware spec issue. I really like the idea of QD, but I'll have to go back to a blank tab if it keeps doing this. Please let me know if I can do anything to help troubleshoot.

**Comment by vertigo220** [comment]
After doing more testing, both with Quick Dial and Bookmark Dial, I've gathered some extra data which can hopefully help speed it up. Since they both seem to be behaving in the same way for the most part, which just some slight variations in speed, I'll just copy my post from the Bookmark Dial issue (https://github.com/sblask/webextension-bookmark-dial/issues/21) I created:

First, let me state I have this same issue with QuickDial. Considering it's loading local data, I would expect the dialer page to load near instantaneously, especially considering my computer's specs, yet it takes a couple seconds. This is with only nine bookmarks, and only one of them has a thumbnail image set. My normal browser is Waterfox, so I tried in Firefox, which only has a few addons installed and none enabled, and it's significantly faster. I disabled most of my addons in Waterfox and it got a lot faster, but still not quite as fast as in Firefox (perhaps due to having more tabs open in Waterfox?). Going through re-enabling them a few at a time didn't reveal any one addon that consistently made much difference on its own, but collectively they do seem to slow it down.

However, I'm not even convinced the addons are to blame. I disabled the background image, using just a solid color, and that made it very fast, similar to Firefox. Thinking maybe it was due to the image being on a platter drive (HDD), I moved it to my SSD, but that didn't make a noticeable difference. In fact, Waterfox, pulling the image from the SSD, was still slower than Firefox which was still using the one on the HDD. It seems to me that Waterfox, possibly due to the addons, is much slower at loading the image, but it doesn't seem like that should be the case. Obviously, the easy solution is to just use a plain background, and that's not a big deal, but it would be nice to be able to use an image without it slowing things down.

Also, as mentioned, Quick Dial suffers from the same problem, though it's a hair slower with the background image and, while it also speeds up without it, the improvement isn't quite as much, so it's still a noticeable delay. So both addons seem to have the same issue, but QD is affected a bit more, though oddly enough QD is slightly faster using the background with most addons disabled. Both are instant on the clean Firefox.
